harness/determined/estimator/_estimator_context.py

```diff
@@ -40,13 +40,21 @@ def __init__(self, *arg: Any, **kwarg: Any) -> None:
 
         self.experimental = EstimatorExperimentalContext(
             env=self.env,
-            hvd_config=self.hvd_config,
             parent=self,
+            distributed_backend=self.distributed_backend
         )
 
+        if self.distributed_backend == "horovod":
```

Comment:
EstimatorTrial can't run with non-horovod distributed backend.  If you ran it with a different dtrain backend, what you have here would silently exhibit weird behavior.  The check you have here follows more closely to the pre-existing code paths, which I assume is why you implemented it this way.  But the pre-existing code paths implicitly assumed that we only ever ran horovod train.  Now that the system no longer enforces that restriction globally, what you have here doesn't make sense in isolation.

As it is right now, if I read this in isolation, I would say, "why does average_training_metrics, something that is common to all dtrain, not apply with to a tensorflow-native distributed backend?  Why only horovod?"

I of course don't want you to add non-horovod support to EstimatorTrial.  I would recommend writing sections like this as `if self.context.distributed.size > 1` (which is equivalent, since only horovod is supported) and I would simply add a check in harness.py that throws an error if somebody tries to launch EstimatorTrial or TFKerasTrial without a horovod launch layer.  With this strategy, there's no need to pass `distributed_backend` into EstimatorTrial or TFKerasTrial at all.